scripts/start.js

```diff
@@ -0,0 +1,126 @@
+const webpack = require('webpack');
+const WebpackDevServer = require('webpack-dev-server');
+const clearConsole = require('react-dev-utils/clearConsole');
+const checkRequiredFiles = require('react-dev-utils/checkRequiredFiles');
+const { choosePort, createCompiler, prepareProxy, prepareUrls } = require('react-dev-utils/WebpackDevServerUtils');
+const paths = require('../config/paths');
+const configFactory = require('../config/webpack.config');
+const createDevServerConfig = require('../config/webpackDevServer.config');
+
+const useYarn = fs.existsSync(paths.yarnLockFile);
+const isInteractive = process.stdout.isTTY;
+
+// Warn and crash if required files are missing
+if (!checkRequiredFiles([paths.appHtml, paths.appIndexJs])) {
+	process.exit(1);
+}
+
+// Tools like Cloud9 rely on this.
+const DEFAULT_PORT = parseInt(process.env.PORT, 10) || 3000;
+const HOST = process.env.HOST || '0.0.0.0';
+
+if (process.env.HOST) {
+	console.log(
+		chalk.cyan(`Attempting to bind to HOST environment variable: ${chalk.yellow(chalk.bold(process.env.HOST))}`)
+	);
+	console.log(`If this was unintentional, check that you haven't mistakenly set it in your shell.`);
+	console.log(`Learn more here: ${chalk.yellow('https://bit.ly/CRA-advanced-config')}`);
+	console.log();
+}
+
+// We require that you explicitly set browsers and do not fall back to
+// browserslist defaults.
+const { checkBrowsers } = require('react-dev-utils/browsersHelper');
+checkBrowsers(paths.appPath, isInteractive)
+	.then(() => {
+		// We attempt to use the default port but if it is busy, we offer the user to
+		// run on a different port. `choosePort()` Promise resolves to the next free port.
+		return choosePort(HOST, DEFAULT_PORT);
+	})
+	.then((port) => {
+		if (port == null) {
+			// We have not found a port.
+			return;
+		}
+		const config = configFactory('development');
+		const protocol = process.env.HTTPS === 'true' ? 'https' : 'http';
+		const appName = require(paths.appPackageJson).name;
+		const useTypeScript = fs.existsSync(paths.appTsConfig);
+		const tscCompileOnError = process.env.TSC_COMPILE_ON_ERROR === 'true';
+		const urls = prepareUrls(protocol, HOST, port);
+		const devSocket = {
+			warnings: (warnings) => devServer.sockWrite(devServer.sockets, 'warnings', warnings),
+			errors: (errors) => devServer.sockWrite(devServer.sockets, 'errors', errors),
+		};
+		// Create a webpack compiler that is configured with custom messages.
+		const compiler = createCompiler({
+			appName,
+			config,
+			devSocket,
+			urls,
+			useYarn,
+			useTypeScript,
+			tscCompileOnError,
+			webpack,
+		});
+		// Load proxy config
+		const proxySetting = require(paths.appPackageJson).proxy;
+		const proxyConfig = prepareProxy(proxySetting, paths.appPublic);
+		// Serve webpack assets generated by the compiler over a web server.
+		const serverConfig = createDevServerConfig(proxyConfig, urls.lanUrlForConfig);
+		const devServer = new WebpackDevServer(compiler, serverConfig);
```

Comment:
i think we should try to remove this at some point (later PR)
because I don't think we can even use this since our app is running within WP